portfolio/src/main/webapp/script.js

```diff
@@ -35,11 +35,65 @@ function addRandomFact() {
     
     //remove this fact and the null from the array
     facts.splice(factIndex,1);
-    console.log(facts.toString());
-    console.log(facts.length);
 
     //update i 
     i = facts.length-1;
   }
+}
+
+
 
+// function to calculate the result of the survey
+function tabulateAnswers() {
+  // initialize variables for each choice's score
+  // If you add more choices and outcomes, you must add another variable here.
+  var c1score = 0;
+  var c2score = 0;
+  var c3score = 0;
+  var c4score = 0;
+  
+  // get a list of the radio inputs on the page
+  var choices = document.getElementsByTagName('input');
+  // loop through all the radio inputs
+  for (i=0; i<choices.length; i++) {
+    // if the radio is checked..
+    if (choices[i].checked) {
+      // add 1 to that choice's score
+      if (choices[i].value == 'c1') {
+        c1score = c1score + 1;
+      }
+      if (choices[i].value == 'c2') {
+        c2score = c2score + 1;
+      }
+      if (choices[i].value == 'c3') {
+        c3score = c3score + 1;
+      }
+      if (choices[i].value == 'c4') {
+        c4score = c4score + 1;
+      }
```

Comment:
Related to the "collect multiples of the same item type" comment, you can simplify this step and not have to copy/paste for each choice group if we have a single container to hold the different choices.